EMMItem.cs

```diff
@@ -119,19 +89,20 @@ public override GlobalItem Clone(Item item, Item itemClone)
 		public override void Load(Item item, TagCompound tag)
 		{
 			GetItemInfo(item).Modifier = Modifier._Load(tag);
+			hasRolled = tag.GetBool("HasRolled");
 
-			// Apply on load. DUH.
 			ModifierContext ctx = new ModifierContext
 			{
-				Method = ModifierContextMethod.OnPickup,
 				Item = item
 			};
 			GetItemInfo(item).Modifier.ApplyItem(ctx);
 		}
 
 		public override TagCompound Save(Item item)
 		{
-			return Modifier.Save(GetItemInfo(item).Modifier);
+			var tag = Modifier.Save(GetItemInfo(item).Modifier);
+			tag.Add("HasRolled", hasRolled);
```

Comment:
HasRolled should be saved in Modifier.Save, not here. See one of my earlier comments. Since it will need access to both Modifier and hasRolled, I suggest we change the static Save so it accepts the item info object instead. (this should also make it easier for future saving changes)